cli/src/lib.rs

```diff
@@ -473,6 +485,84 @@ fn domains(configuration: &Configuration) -> [Domain; 1] {
     [Domain::from(GenesisDomain::new(key))]
 }
 
+pub mod style {
+    //! Style and colouration of Iroha CLI outputs.
+    use owo_colors::{OwoColorize, Style};
+
+    /// Styling information set at run-time for pretty-printing with colour
+    #[derive(Clone, Copy, Debug)]
+    pub struct Styling {
+        /// Positive highlight
+        pub positive: Style,
+        /// Negative highlight. Usually error message.
+        pub negative: Style,
+        /// Neutral highlight
+        pub highlight: Style,
+        /// Minor message
+        pub minor: Style,
+    }
+
+    impl Default for Styling {
+        fn default() -> Self {
+            Self {
+                positive: Style::new().green().bold(),
+                negative: Style::new().red().bold(),
+                highlight: Style::new().bold(),
+                minor: Style::new().green(),
+            }
+        }
+    }
+
+    /// Determine if message colourisation is to be enabled
+    pub fn should_enable_colour() -> bool {
+        supports_color::on(supports_color::Stream::Stdout).is_none()
+            || std::env::var("TERMINAL_COLORS")
+                .map(|s| !s.as_str().parse().unwrap_or(true))
+                .unwrap_or(false)
+    }
+
+    impl Styling {
+        #[must_use]
+        /// Constructor
+        pub fn new() -> Self {
+            if should_enable_colour() {
+                Self::no_color()
+            } else {
+                Self::default()
+            }
+        }
+
+        fn no_color() -> Self {
+            Self {
+                positive: Style::new(),
+                negative: Style::new(),
+                highlight: Style::new(),
+                minor: Style::new(),
+            }
+        }
+
+        /// Produce documentation for argument group
+        pub fn or(&self, arg_group: &[&str; 2]) -> String {
+            format!(
+                "`{}` (short `{}`)",
+                arg_group[0].style(self.positive),
+                arg_group[1].style(self.minor)
+            )
+        }
+
+        /// Convenience method for ".json or .json5" pattern
+        pub fn json_ext(&self, name: &str) -> String {
```

Comment:
```suggestion
        pub fn with_json_ext(&self, name: &str) -> String {
```
This could be more easily understable. Otherwise it reads almost inverted.